AngularJS/components/reports/reportCRUD.js

```diff
@@ -0,0 +1,141 @@
+function ReportListController(reportService) {
+    //Lists all reports in tabular form, with add, edit and remove options against each report item.
+    var $ctrl = this;
+    $ctrl.master = {};
+    $ctrl.report = {};
+    $ctrl.isUpdate = false;
+    $ctrl.current_report_item = {};
+    $ctrl.types = [
+        {
+            keys: "AN",
+            value: "Annual"
+        }, 
+        {
+            keys: "MO",
+            value: "Monthly"
+        }, 
+        {
+            keys: "WE",
+            value: "Weekly"
+        }
+    ];
+
+    $ctrl.$onInit(){
+        get_token();
```

Comment:
token should be setup on start of the app module it should not depend on some component that might not start when on a different url.